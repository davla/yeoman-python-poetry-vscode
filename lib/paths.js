import path from "path";
import { fileURLToPath } from "url";

export const moduleDirName = (meta) => path.dirname(fileURLToPath(meta.url));

export const rootPath = path.join(moduleDirName(import.meta), "..");
