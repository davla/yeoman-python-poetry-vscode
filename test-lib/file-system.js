import fs from "node:fs/promises";
import path from "node:path";

const fileInCwd = (runResult, fileName) =>
  path.join(runResult.cwd ?? runResult, fileName);

export const readFileInCwd = (runResult, fileName) =>
  fs.readFile(fileInCwd(runResult, fileName), "utf-8");

export const writeFileInCwd = (runResult, fileName, content) =>
  fs.writeFile(fileInCwd(runResult, fileName), content);
