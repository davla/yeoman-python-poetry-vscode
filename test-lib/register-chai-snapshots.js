import chai from "chai";
import { jestSnapshotPlugin as chaiSnapshot } from "mocha-chai-jest-snapshot";

chai.use(chaiSnapshot());
