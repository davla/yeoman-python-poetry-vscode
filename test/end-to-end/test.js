import fs from "node:fs/promises";
import path from "node:path";

import { execa } from "execa";

import { moduleDirName } from "../../lib/paths.js";
import { readCwd } from "../lib/file-system.js";

import "../lib/register-chai-snapshots.js";

const endToEndPath = path.join(
  moduleDirName(import.meta),
  "test",
  "end-to-end",
);

describe("end-to-end tests", () => {
  beforeEach(function () {
    this.prepare = async function (subGeneratorName = null) {
      this.outputPath = path.join(endToEndPath, subGeneratorName ?? "app");
      await fs.mkdir(this.outputPath, { recursive: true });

      this.generatorName = "python-poetry-vscode";
      if (subGeneratorName !== null) {
        this.generatorName += ":" + subGeneratorName;
      }
    };

    this.execTest = async function (parameters = {}) {
      const cliArgs = Object.entries(parameters).flatMap(([key, value]) => [
        "--" + key,
        value,
      ]);
      await execa("npx", ["yo", this.generatorName, ...cliArgs], {
        cwd: this.outputPath,
      });

      (await readCwd(this.outputPath)).should.matchSnapshot();
    };
  });

  afterEach(async function () {
    await fs.rm(this.outputPath, { recursive: true });
  });

  it.skip("python-poetry-vscode", async function () {
    await this.prepare();
    await this.execTest({
      "author-email": "jin.kazama@tekken.jp",
      "author-name": "Jin Kazama",
      description: "I don't actually like fighting games",
      license: "GPL-3.0",
      "package-name": "tekken-3",
      "package-version": "0.5.3",
      "python-version": "^3.10.2",
      repository: "https://github.com/jin-kazama/tekken-3",
    });
  });

  it("python-poetry-vscode:poetry", async function () {
    await this.prepare("poetry");
    await this.execTest({
      "author-email": "paul.phoenix@tekken.us",
      "author-name": "Paul Phoenix",
      description: "I don't actually like fighting games",
      license: "MIT",
      "package-name": "tekken",
      "package-version": "2.0.2",
      "python-version": "^3.10.0",
      repository: "https://github.com/paul-phoenix/tekken",
    });
  });

  it("python-poetry-vscode:python-package", async function () {
    await this.prepare("python-package");
    await this.execTest({
      "package-name": "yoyo",
      "package-version": "1.9.0",
    });
  });

  it("python-poetry-vscode:vscode", async function () {
    await this.prepare("vscode");
    await this.execTest();
  });
});
