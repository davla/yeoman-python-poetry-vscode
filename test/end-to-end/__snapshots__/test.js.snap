// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`end-to-end tests python-poetry-vscode:poetry 1`] = `
Object {
  "pyproject.toml": Object {
    "build-system": Object {
      "build-backend": "poetry.core.masonry.api",
      "requires": Array [
        "poetry-core",
      ],
    },
    "tool": Object {
      "poetry": Object {
        "authors": Array [
          "Paul Phoenix <paul.phoenix@tekken.us>",
        ],
        "dependencies": Object {
          "python": "^3.10.0",
        },
        "description": "I don't actually like fighting games",
        "license": "MIT",
        "name": "tekken",
        "repository": "https://github.com/paul-phoenix/tekken",
        "version": "2.0.2",
      },
    },
  },
}
`;

exports[`end-to-end tests python-poetry-vscode:python-package 1`] = `
Object {
  "tests/__init__.py": "",
  "tests/test_yoyo.py": "from yoyo import __version__


def test_version():
    assert __version__ == \\"1.9.0\\"
",
  "yoyo/__init__.py": Object {
    "__version__": "1.9.0",
  },
}
`;

exports[`end-to-end tests python-poetry-vscode:vscode 1`] = `
Object {
  "poetry.toml": Object {
    "virtualenvs": Object {
      "in-project": true,
    },
  },
  "pyproject.toml": Object {
    "tool": Object {
      "poetry": Object {
        "dependencies": Object {
          "black": "^23.3.0",
          "isort": "^5.12.0",
        },
        "group": Object {
          "dev": Object {
            "dependencies": Object {
              "pytest": "^7.3.1",
            },
          },
        },
      },
    },
  },
}
`;
